listener.patch

```diff
@@ -0,0 +1,89 @@
+diff --git a/collector/internal/telemetryapi/listener.go b/collector/internal/telemetryapi/listener.go
+index 8499d4e..93ff0dc 100644
+--- a/collector/internal/telemetryapi/listener.go
++++ b/collector/internal/telemetryapi/listener.go
+@@ -17,18 +17,32 @@ package telemetryapi
+ import (
+ 	"context"
+ 	"encoding/json"
++	"errors"
+ 	"fmt"
+ 	"io"
++	"math/rand"
++	"net"
+ 	"net/http"
+ 	"os"
++	"syscall"
+ 	"time"
+ 
+ 	"github.com/golang-collections/go-datastructures/queue"
+ 	"go.uber.org/zap"
+ )
+ 
+-const defaultListenerPort = "53612"
+-const initialQueueSize = 5
++const (
++	initialQueueSize = 5
++	maxRetries       = 5
++	// Define ephemeral port range (typical range is 49152-65535)
++	minPort = 49152
++	maxPort = 65535
++)
++
++// getRandomPort returns a random port number within the ephemeral range
++func getRandomPort() string {
++	return fmt.Sprintf("%d", rand.Intn(maxPort-minPort)+minPort)
++}
+ 
+ // Listener is used to listen to the Telemetry API
+ type Listener struct {
+@@ -46,21 +60,44 @@ func NewListener(logger *zap.Logger) *Listener {
+ 	}
+ }
+ 
+-func listenOnAddress() string {
++func (s *Listener) tryBindPort() (string, error) {
++	for i := 0; i < maxRetries; i++ {
++		port := getRandomPort()
++		address := listenOnAddress(port)
++
++		l, err := net.Listen("tcp", address)
++		if err != nil {
++			if errors.Is(err, syscall.EADDRINUSE) {
++				s.logger.Debug("Port in use, trying another",
++					zap.String("address", address))
++				continue
++			}
++			return "", err
++		}
++		l.Close()
```

Comment:
I think this needs to return a [`net.Listener`](https://pkg.go.dev/net#Listener) that can be handed to [`s.httpServer.Serve()` ](https://pkg.go.dev/net/http#Server.Serve) so that once a valid port is acquired there is no chance that something else starts using that port between the call to `l.Close()` here and later starting a server on that port.